x-pack/plugin/ql/src/main/java/org/elasticsearch/xpack/ql/async/AsyncTaskManagementService.java

```diff
@@ -234,7 +236,15 @@ private ActionListener<Response> wrapStoringListener(
                 }
             } else {
                 // We finished after timeout - saving results
-                storeResults(searchTask, new StoredAsyncResponse<>(response, threadPool.absoluteTimeInMillis() + keepAlive.getMillis()));
+                storeResults(
+                    searchTask,
+                    new StoredAsyncResponse<>(response, threadPool.absoluteTimeInMillis() + keepAlive.getMillis()),
+                    ActionListener.running(() -> {
+                        if (response instanceof Releasable releasable) {
+                            releasable.close();
```

Comment:
@dnhatn this is kinda part of the problem, and part of the solution. We cannot close here, if there is an async get on the task already waiting for the response.